src/main/java/org/opensearch/knn/index/SpaceType.java

```diff
@@ -62,7 +62,10 @@ public float scoreToDistanceTranslation(float score) {
     COSINESIMIL("cosinesimil") {
         @Override
         public float scoreTranslation(float rawScore) {
-            return 1 / (1 + rawScore);
+            // To be consistent, we will be using same formula used by lucene as mentioned below
+            // https://github.com/apache/lucene/blob/0494c824e0ac8049b757582f60d085932a890800/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java#L73
+            // Here raw score = 1 - cosine(x,y), hence, formula will be updated to below to get final score as similar to Lucene.
```

Comment:
I think we need to make this explicit in the contract - maybe add to java doc API. Then, if an engine does something else, it needs to be overridden in the engine specific space translations (see the Faiss specific engine methods for more details).

Reply:
Updated